repo2docker/buildpacks/base.py

```diff
@@ -414,12 +414,26 @@ def get_start_script(self):
         """
         return None
 
+    @property
+    def binder_dir(self):
+        has_binder = os.path.isdir("binder")
+        has_dotbinder = os.path.isdir(".binder")
+
+        if has_binder and has_dotbinder:
+            raise RuntimeError(
+                "The repository contains both a 'binder' and a '.binder' "
+                "directory. However they are exclusive.")
```

Comment:
It may make sense to raise an informative warning here instead. This implementation will render a repo unbuildable if both directories exist. The warning would just say something like: "We've found both a `.binder` and `binder` directory in your repo, we're only going to use `binder`".

Reply:
I think repo2docker should stop running if there are two, not warn and continue. My feeling is that it is almost certainly an oversight on the users side when this happens. Combined with the huge amount of output repo2docker already generates my hunch is that most people will miss the warning. Can we think of a reason why a user would want to have both at the same time if they know that they are exclusive?

Reply:
I'm unaware of how an error would propegate to MyBinder.org etc, would this be easy to get for the users of mybinder.org if it fails like this? What would happen if we try to make it a warning for users of mybinder.org or binderhubs rather?

Reply:
As long as we can propagate the error to the user running their session on a binderhub, I'm +1 on making this error rather than warn, I think we could get into some confusing situations otherwise (partially because remembering which one takes precedence will be totally arbitrary and add cognitive burden...I think it's easier to just remember to use one or the other but not both)

Reply:
You will get a failed build on a BinderHub with (I think) the traceback printed as the last thing in the logs. This can already happen today for example with https://mybinder.org/v2/git/https%3A%2F%2Fgithub.com%2Fbetatim%2Fdsgsdgdsg.git/95f2eecf43686d2a45e7c15452b99e9d6346295c (the revision doesn't exist). (We could make it a nicer error message by adding some `try ... except` handling at the highest level of the repo2docker CLI app, but that is a new PR.)

Reply:
Okay, it sounds like there in consensus here.